### ISSUE TITLE
Login with Device Code and activated Conditional Access does not work

### ISSUE THREAD

**Issue by J0F3** [Azure/azure-powershell#9427]
## Description
When a Conditional Access Policy is configured in the Azure AD which requires for example MFA the Connect-AzAccount fails on PowerShell Core 6 or whenever it uses the Device Login/Code workflow. 
Actually not sure if this should be the intended result or if this is a bug in the Module or elsewhere. 

## Steps to reproduce

1. Configure an Conditional Access Policy in the Azure AD to enforce MFA (for example) for all Apps or only the "Microsoft Azure Management" App. 
Basically follow this guide: https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/app-based-mfa

2. Run Connect-AzAccount in PowerShell Core 6

```powershell
Connect-AzAccount
```

## Environment data

```

Name                           Value
----                           -----
PSVersion                      6.2.1
PSEdition                      Core
GitCommitId                    6.2.1
OS                             Microsoft Windows 10.0.18362 
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

## Module versions

```powershell


    Directory: C:\Users\Jonas Feller\Documents\PowerShell\Modules

ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     2.2.0      Az                                  Core,Desk 
Script     1.5.2      Az.Accounts                         Core,Desk {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosaveΓÇª}
Script     1.0.1      Az.Aks                              Core,Desk {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAksCredentialΓÇª}
Script     1.1.0      Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServerΓÇª}
Script     1.1.0      Az.ApiManagement                    Core,Desk {Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion, Add-AzApiManagementUserToGroupΓÇª}
Script     1.0.0      Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, New-AzApplicationInsights, Remove-AzApplicationInsights, Set-AzApplicationInsightsPricingPlanΓÇª}
Script     1.2.2      Az.Automation                       Core,Desk {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDscNodeConfigurationΓÇª}
Script     1.1.0      Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccountΓÇª}
Script     1.0.0      Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount, Get-AzConsumptionBudgetΓÇª}
Script     1.3.0      Az.Cdn                              Core,Desk {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfileΓÇª}
Script     1.1.1      Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSku, Get-AzCognitiveServicesAccountTypeΓÇª}
Script     2.2.0      Az.Compute                          Core,Desk {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySetΓÇª}
Script     1.0.1      Az.ContainerInstance                Core,Desk {New-AzContainerGroup, Get-AzContainerGroup, Remove-AzContainerGroup, Get-AzContainerInstanceLog}
Script     1.0.1      Az.ContainerRegistry                Core,Desk {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistryΓÇª}
Script     1.1.1      Az.DataFactory                      Core,Desk {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2ΓÇª}
Script     1.0.0      Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCredential, Set-AzDataLakeAnalyticsCatalogCredentialΓÇª}
Script     1.2.1      Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLakeStoreTrustedIdProviderΓÇª}
Script     1.0.0      Az.DeploymentManager                Core,Desk {Get-AzDeploymentManagerArtifactSource, New-AzDeploymentManagerArtifactSource, Set-AzDeploymentManagerArtifactSource, Remove-AzDeploymentManagerArtifactSourceΓÇª}
Script     1.0.0      Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicyΓÇª}
Script     1.1.0      Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSetΓÇª}
Script     1.1.1      Az.EventGrid                        Core,Desk {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKeyΓÇª}
Script     1.2.0      Az.EventHub                         Core,Desk {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespaceΓÇª}
Script     1.0.0      Az.FrontDoor                        Core,Desk {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoorΓÇª}
Script     2.0.0      Az.HDInsight                        Core,Desk {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinitionΓÇª}
Script     1.1.0      Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJobΓÇª}
Script     1.2.0      Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertificateOperationΓÇª}
Script     1.2.1      Az.LogicApp                         Core,Desk {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountAssembly, Get-AzIntegrationAccountBatchConfiguration, Get-AzIntegrationAccountCallbackUrlΓÇª}
Script     1.1.0      Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlCommitmentPlanΓÇª}
Script     1.0.0      Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}
Script     1.1.0      Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKey, Set-AzMediaServiceKey, Get-AzMediaServiceKey, Get-AzMediaServiceNameAvailabilityΓÇª}
Script     1.2.1      Az.Monitor                          Core,Desk {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfileΓÇª}
Script     1.9.0      Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayAuthenticationCertificate, Remove-AzAppliΓÇª
Script     1.1.0      Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRule, Get-AzNotificationHubListKey, Get-AzNotificationHubPNSCredentialΓÇª}
Script     1.2.0      Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOperationalInsightsLinuxCustomLogCollection, Disable-AzOpeΓÇª
Script     1.1.1      Az.PolicyInsights                   Core,Desk {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediationΓÇª}
Script     1.1.0      Az.PowerBIEmbedded                  Core,Desk {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKey, Get-AzPowerBIWorkspaceΓÇª}
Script     1.4.1      Az.RecoveryServices                 Core,Desk {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServicesVaultΓÇª}
Script     1.1.0      Az.RedisCache                       Core,Desk {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchScheduleΓÇª}
Script     1.0.1      Az.Relay                            Core,Desk {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespaceΓÇª}
Script     1.4.0      Az.Resources                        Core,Desk {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignmentΓÇª}
Script     1.2.0      Az.ServiceBus                       Core,Desk {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespaceΓÇª}
Script     1.1.0      Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricApplicationCertificate, Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertificate, Add-AzServiceFabricNodeΓÇª}
Script     1.0.2      Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKeyΓÇª}
Script     1.11.0     Az.Sql                              Core,Desk {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseUpgradeHinΓÇª
Script     1.3.0      Az.Storage                          Core,Desk {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKeyΓÇª}
Script     1.0.0      Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsFunction, Get-AzStreamAnalyticsDefaultFunctionDefinition, New-AzStreamAnalyticsFunction, Remove-AzStreamAnalyticsFunctionΓÇª}
Script     1.0.1      Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerCustomHeaderToProfile, Remove-AzTrafficManagerCustomHeΓÇª
Script     1.2.2      Az.Websites                         Core,Desk {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlanΓÇª}
Script     1.4.1      PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSourceΓÇª}
Script     2.1.4      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapabilityΓÇª}

    Directory: C:\program files\powershell\6\Modules

ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Manifest   6.1.0.0    CimCmdlets                          Core      {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSessionΓÇª}
Manifest   1.2.3.0    Microsoft.PowerShell.Archive        Desk      {Compress-Archive, Expand-Archive}
Manifest   6.1.0.0    Microsoft.PowerShell.Diagnostics    Core      {Get-WinEvent, New-WinEvent}
Manifest   6.1.0.0    Microsoft.PowerShell.Host           Core      {Start-Transcript, Stop-Transcript}
Manifest   6.1.0.0    Microsoft.PowerShell.Management     Core      {Add-Content, Clear-Content, Clear-ItemProperty, Join-PathΓÇª}
Manifest   6.1.0.0    Microsoft.PowerShell.Security       Core      {Get-Acl, Set-Acl, Get-PfxCertificate, Get-CredentialΓÇª}
Manifest   6.1.0.0    Microsoft.PowerShell.Utility        Core      {Export-Alias, Get-Alias, Import-Alias, New-AliasΓÇª}
Manifest   6.1.0.0    Microsoft.WSMan.Management          Core      {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP, Set-WSManQuickConfigΓÇª}
Script     1.3.2      PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSourceΓÇª}
Script     2.1.3      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapabilityΓÇª}
Script     0.0        PSDesiredStateConfiguration         Desk      {GetSyntax, Get-CompatibleVersionAddtionaPropertiesStr, ImportClassResourcesFromModule, Write-NodeMOFFileΓÇª}
Script     6.1.0.0    PSDiagnostics                       Core      {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManTrace, Enable-PSTraceΓÇª}
Script     2.0.0      PSReadLine                          Desk      {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remove-PSReadLineKeyHandler, Get-PSReadLineOptionΓÇª}
Binary     1.1.2      ThreadJob                           Desk      Start-ThreadJob

    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules

ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Manifest   1.0.0.0    AppBackgroundTask                   Core,Desk {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgroundTaskDiagnosticLog, Set-AppBackgroundTaskResourcePolicy, Unregister-AppBackgroundTaskΓÇª}
Manifest   2.0.1.0    Appx                                Core,Desk {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest, Remove-AppxPackageΓÇª}
Script     1.0.0.0    AssignedAccess                      Core,Desk {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAccess}
Manifest   1.0.0.0    BitLocker                           Core,Desk {Unlock-BitLocker, Suspend-BitLocker, Resume-BitLocker, Remove-BitLockerKeyProtectorΓÇª}
Manifest   1.0.0.0    BranchCache                         Core,Desk {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disable-BCDowngradingΓÇª}
Manifest   1.0        Defender                            Core,Desk {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remove-MpPreferenceΓÇª}
Manifest   1.0.2.0    DeliveryOptimization                Core,Desk {Delete-DeliveryOptimizationCache, Get-DeliveryOptimizationStatus, Set-DeliveryOptimizationStatus, Get-DeliveryOptimizationPerfSnapΓÇª}
Manifest   1.0.0.0    DirectAccessClientComponents        Core,Desk {Disable-DAManualEntryPointSelection, Enable-DAManualEntryPointSelection, Get-DAClientExperienceConfiguration, Get-DAEntryPointTableItemΓÇª}
Script     3.0        Dism                                Core,Desk {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-WindowsCapability, Add-WindowsImageΓÇª}
Manifest   1.0.0.0    DnsClient                           Core,Desk {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-DnsClientCacheΓÇª}
Manifest   1.0.0.0    EventTracingManagement              Core,Desk {Start-EtwTraceSession, New-EtwTraceSession, Get-EtwTraceSession, Update-EtwTraceSessionΓÇª}
Manifest   2.0.0.0    International                       Core,Desk {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMethodOverride, Get-WinHomeLocation, Set-WinHomeLocationΓÇª}
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    Core,Desk {Get-WinEvent, Get-Counter, Import-Counter, Export-CounterΓÇª}
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  Core,Desk {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser, Get-LocalGroupΓÇª}
Manifest   3.1.0.0    Microsoft.PowerShell.Management     Core,Desk {Add-Content, Clear-Content, Clear-ItemProperty, Join-PathΓÇª}
Manifest   1.0        MMAgent                             Core,Desk {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgentΓÇª}
Manifest   2.0.0.0    NetAdapter                          Core,Desk {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsulatedPacketTaskOffloadΓÇª}
Manifest   1.0.0.0    NetConnection                       Core,Desk {Get-NetConnectionProfile, Set-NetConnectionProfile}
Manifest   1.0.0.0    NetEventPacketCapture               Core,Desk {New-NetEventSession, Remove-NetEventSession, Get-NetEventSession, Set-NetEventSessionΓÇª}
Manifest   2.0.0.0    NetLbfo                             Core,Desk {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam, Get-NetLbfoTeamMemberΓÇª}
Manifest   1.0.0.0    NetNat                              Core,Desk {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMapping, Get-NetNatSessionΓÇª}
Manifest   2.0.0.0    NetQos                              Core,Desk {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, New-NetQosPolicy}
Manifest   2.0.0.0    NetSecurity                         Core,Desk {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryptoProposalΓÇª}
Manifest   1.0.0.0    NetSwitchTeam                       Core,Desk {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam, Rename-NetSwitchTeamΓÇª}
Manifest   1.0.0.0    NetTCPIP                            Core,Desk {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol, Get-NetIPv6ProtocolΓÇª}
Manifest   1.0.0.0    NetworkConnectivityStatus           Core,Desk {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset-NCSIPolicyConfiguration, Set-NCSIPolicyConfiguration}
Manifest   1.0.0.0    NetworkSwitchManager                Core,Desk {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEthernetPort, Get-NetworkSwitchEthernetPort, Remove-NetworkSwitchEthernetPortIPAddressΓÇª}
Manifest   1.0.0.0    NetworkTransition                   Core,Desk {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfiguration, Disable-NetIPHttpsProfile, Disable-NetNatTransitionConfigurationΓÇª}
Manifest   1.0.0.0    PcsvDevice                          Core,Desk {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart-PcsvDeviceΓÇª}
Manifest   1.0.0.0    PnpDevice                           Core,Desk {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Disable-PnpDevice}
Manifest   1.1        PrintManagement                     Core,Desk {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-PrintConfigurationΓÇª}
Binary     1.0.11     ProcessMitigations                  Core,Desk {Get-ProcessMitigation, Set-ProcessMitigation, ConvertTo-ProcessMitigationPolicy}
Manifest   1.0.0.0    ScheduledTasks                      Core,Desk {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTask, Unregister-ScheduledTaskΓÇª}
Manifest   2.0.0.0    SecureBoot                          Core,Desk {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBootUEFI, Format-SecureBootUEFIΓÇª}
Manifest   2.0.0.0    SmbShare                            Core,Desk {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShareAccessΓÇª}
Manifest   2.0.0.0    SmbWitness                          Core,Desk {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbwΓÇª}
Manifest   1.0.0.0    StartLayout                         Core,Desk {Export-StartLayout, Import-StartLayout, Export-StartLayoutEdgeAssets, Get-StartApps}
Manifest   2.0.0.0    Storage                             Core,Desk {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-PhysicalDisk, Add-StorageFaultDomainΓÇª}
Manifest   1.0.0.0    TroubleshootingPack                 Core,Desk {Get-TroubleshootingPack, Invoke-TroubleshootingPack}
Manifest   2.0.0.0    TrustedPlatformModule               Core,Desk {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-TpmΓÇª}
Binary     2.1.639.0  UEV                                 Core,Desk 
Manifest   2.0.0.0    VpnClient                           Core,Desk {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection, Get-VpnConnectionΓÇª}
Manifest   1.0.0.0    Wdac                                Core,Desk {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsnΓÇª}
Manifest   2.0.0.0    Whea                                Core,Desk {Get-WheaMemoryPolicy, Set-WheaMemoryPolicy}
Manifest   1.0.0.0    WindowsDeveloperLicense             Core,Desk {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLicense, Show-WindowsDeveloperLicenseRegistration}
Script     1.0        WindowsErrorReporting               Core,Desk {Enable-WindowsErrorReporting, Disable-WindowsErrorReporting, Get-WindowsErrorReporting}
Manifest   1.0.0.0    WindowsSearch                       Core,Desk {Get-WindowsSearchSetting, Set-WindowsSearchSetting}
Manifest   1.0.0.0    WindowsUpdate                       Core,Desk Get-WindowsUpdateLog
Manifest   1.0.0.2    WindowsUpdateProvider               Core,Desk {Get-WUAVersion, Get-WULastInstallationDate, Get-WULastScanSuccessDate, Get-WUIsPendingRebootΓÇª}
```

## Debug output

```
Connect-AzAccount
DEBUG: 10:18:42 - ConnectAzureRmAccountCommand begin processing with ParameterSet 'UserWithSubscriptionId'.
DEBUG: 10:18:42 - Autosave setting from startup session: 'CurrentUser'
DEBUG: 10:18:42 - No autosave setting detected in environment variable 'AzContextAutoSave'.
DEBUG: 10:18:42 - Using Autosave scope 'CurrentUser'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: 'Common', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://management.core.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/Common/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: 'Common', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob
WARNING: To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code XXXXXXXX to authenticate.
DEBUG: [ADAL]: Information: 2019-06-14T08:18:42.9947861Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2019-06-14T08:18:42.9949884Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2019-06-14T08:18:42.9952865Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: User
        , Authority Host: login.microsoftonline.com

DEBUG: [ADAL]: Information: 2019-06-14T08:18:42.9954785Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: === Token Acquisition started:
        Authority: https://login.microsoftonline.com/Common/
        Resource: https://management.core.windows.net/
        ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
        CacheType: null
        Authentication Target: User


DEBUG: [ADAL]: Error: 2019-06-14T08:19:09.4882703Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: Exception type: Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException, ErrorCode: invalid_grant, StatusCode: 401 ---> Exception type: System.Net.Http.HttpRequestException ---> Exception type: Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException, ErrorCode: {"error":"invalid_grant","error_description":"AADSTS50097: Device authentication is required.\r\nTrace ID: e0e6689e-7e5d-43b3-be28-a43490025900\r\nCorrelation ID: 01e262f5-9585-45b8-9820-2a12448900fb\r\nTimestamp: 2019-06-14 08:19:08Z","error_codes":[50097],"timestamp":"2019-06-14 08:19:08Z","trace_id":"e0e6689e-7e5d-43b3-be28-a43490025900","correlation_id":"01e262f5-9585-45b8-9820-2a12448900fb"}
--- End of inner exception stack trace ---
--- End of inner exception stack trace ---
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T](Boolean respondToDeviceAuthChallenge)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T]()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendHttpMessageAsync(IRequestParameters requestParameters)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenByDeviceCodeHandler.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.CheckAndAcquireTokenUsingBrokerAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.RunAsync()

DEBUG: [ADAL]: Error: 2019-06-14T08:19:09.4891608Z: 24431be9-2ca5-4d4c-a98e-0d564b7df532 - LoggerBase.cs: Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException: AADSTS50097: Device authentication is required.
Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
Timestamp: 2019-06-14 08:19:08Z ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 (Unauthorized). ---> Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException: {"error":"invalid_grant","error_description":"AADSTS50097: Device authentication is required.\r\nTrace ID: e0e6689e-7e5d-43b3-be28-a43490025900\r\nCorrelation ID: 01e262f5-9585-45b8-9820-2a12448900fb\r\nTimestamp: 2019-06-14 08:19:08Z","error_codes":[50097],"timestamp":"2019-06-14 08:19:08Z","trace_id":"e0e6689e-7e5d-43b3-be28-a43490025900","correlation_id":"01e262f5-9585-45b8-9820-2a12448900fb"}: Unknown error
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T](Boolean respondToDeviceAuthChallenge)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T]()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendHttpMessageAsync(IRequestParameters requestParameters)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenByDeviceCodeHandler.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.CheckAndAcquireTokenUsingBrokerAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.RunAsync()
        ErrorCode: invalid_grant
        StatusCode: 401

Connect-AzAccount : AADSTS50097: Device authentication is required.
Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
Timestamp: 2019-06-14 08:19:08Z: Response status code does not indicate success: 401 (Unauthorized).
At line:1 char:1
+ Connect-AzAccount
+ ~~~~~~~~~~~~~~~~~
+ CategoryInfo          : CloseError: (:) [Connect-AzAccount], AadAuthenticationFailedException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand

DEBUG: AzureQoSEvent: CommandName - Connect-AzAccount; IsSuccess - False; Duration - 00:00:27.5510995; Exception - Microsoft.Azure.Commands.Common.Authentication.AadAuthenticationFailedException: AADSTS50097: Device authentication is required.
Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
Timestamp: 2019-06-14 08:19:08Z: Response status code does not indicate success: 401 (Unauthorized). ---> Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException: AADSTS50097: Device authentication is required.
Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
Timestamp: 2019-06-14 08:19:08Z ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 (Unauthorized). ---> Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException: {"error":"invalid_grant","error_description":"AADSTS50097: Device authentication is required.\r\nTrace ID: e0e6689e-7e5d-43b3-be28-a43490025900\r\nCorrelation ID: 01e262f5-9585-45b8-9820-2a12448900fb\r\nTimestamp: 2019-06-14 08:19:08Z","error_codes":[50097],"timestamp":"2019-06-14 08:19:08Z","trace_id":"e0e6689e-7e5d-43b3-be28-a43490025900","correlation_id":"01e262f5-9585-45b8-9820-2a12448900fb"}: Unknown error
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T](Boolean respondToDeviceAuthChallenge)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T]()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendHttpMessageAsync(IRequestParameters requestParameters)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenByDeviceCodeHandler.SendTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.CheckAndAcquireTokenUsingBrokerAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.RunAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.AcquireTokenByDeviceCodeAsync(DeviceCodeResult deviceCodeResult)
   at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.DoAcquireToken(AdalConfiguration config, String userId, SecureString password, Action`1 promptAction, Boolean renew)
   at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.SafeAquireToken(AdalConfiguration config, Action`1 promptAction, String userId, SecureString password, Exception& ex)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.AcquireToken(AdalConfiguration config, Action`1 promptAction, String userId, SecureString password, Boolean renew)
   at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.GetAccessToken(AdalConfiguration config, String promptBehavior, Action`1 promptAction, String userId, SecureString password, String credentialType)
   at Microsoft.Azure.Commands.Common.Authentication.AdalTokenProvider.GetAccessToken(AdalConfiguration config, String promptBehavior, Action`1 promptAction, String userId, SecureString password, String credentialType)
   at Microsoft.Azure.Commands.Common.Authentication.Factories.AuthenticationFactory.Authenticate(IAzureAccount account, IAzureEnvironment environment, String tenant, SecureString password, String promptBehavior, Action`1 promptAction, IAzureTokenCache tokenCache, String resourceId)
   at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.AcquireAccessToken(IAzureAccount account, IAzureEnvironment environment, String tenantId, SecureString password, String promptBehavior, Action`1 promptAction)
   at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.ListAccountTenants(IAzureAccount account, IAzureEnvironment environment, SecureString password, String promptBehavior, Action`1 promptAction)
   at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.Login(IAzureAccount account, IAzureEnvironment environment, String tenantId, String subscriptionId, String subscriptionName, SecureString password, Boolean skipValidation, Action`1 promptAction, String name, Boolean shouldPopulateContextList)
   at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.<>c__DisplayClass92_0.<ExecuteCmdlet>b__0(AzureRmProfile localProfile, RMProfileClient profileClient, String name)
   at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.<>c__DisplayClass94_0.<SetContextWithOverwritePrompt>b__0(AzureRmProfile prof, RMProfileClient client)
   at Microsoft.Azure.Commands.Profile.Common.AzureContextModificationCmdlet.ModifyContext(Action`2 contextAction)
   at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.SetContextWithOverwritePrompt(Action`3 setContextAction)
   at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 10:19:09 - ConnectAzureRmAccountCommand end processing.
DEBUG: 10:19:09 - ConnectAzureRmAccountCommand end processing.
```

## Error output

```
DEBUG: 10:20:43 - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.


   HistoryId: 2
Message        : AADSTS50097: Device authentication is required.
                 Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
                 Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
                 Timestamp: 2019-06-14 08:19:08Z: Response status code does not indicate success: 401 (Unauthorized).
StackTrace     :    at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.AcquireToken(AdalConfiguration config, Action`1 promptAction, String userId, SecureString password, Boolean renew)
                    at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.GetAccessToken(AdalConfiguration config, String promptBehavior, Action`1 promptAction, String userId, SecureString password, String credentialType)
                    at Microsoft.Azure.Commands.Common.Authentication.AdalTokenProvider.GetAccessToken(AdalConfiguration config, String promptBehavior, Action`1 promptAction, String userId, SecureString password, String credentialType)
                    at Microsoft.Azure.Commands.Common.Authentication.Factories.AuthenticationFactory.Authenticate(IAzureAccount account, IAzureEnvironment environment, String tenant, SecureString password, String promptBehavior,
                 Action`1 promptAction, IAzureTokenCache tokenCache, String resourceId)
                    at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.AcquireAccessToken(IAzureAccount account, IAzureEnvironment environment, String tenantId, SecureString password, String promptBehavior, Action`1
                 promptAction)
                    at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.ListAccountTenants(IAzureAccount account, IAzureEnvironment environment, SecureString password, String promptBehavior, Action`1 promptAction)
                    at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.Login(IAzureAccount account, IAzureEnvironment environment, String tenantId, String subscriptionId, String subscriptionName, SecureString password,
                 Boolean skipValidation, Action`1 promptAction, String name, Boolean shouldPopulateContextList)
                    at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.<>c__DisplayClass92_0.<ExecuteCmdlet>b__0(AzureRmProfile localProfile, RMProfileClient profileClient, String name)
                    at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.<>c__DisplayClass94_0.<SetContextWithOverwritePrompt>b__0(AzureRmProfile prof, RMProfileClient client)
                    at Microsoft.Azure.Commands.Profile.Common.AzureContextModificationCmdlet.ModifyContext(Action`2 contextAction)
                    at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.SetContextWithOverwritePrompt(Action`3 setContextAction)
                    at Microsoft.Azure.Commands.Profile.ConnectAzureRmAccountCommand.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : Microsoft.Azure.Commands.Common.Authentication.AadAuthenticationFailedException
InvocationInfo : {Connect-AzAccount}
Line           : Connect-AzAccount
Position       : At line:1 char:1
                 + Connect-AzAccount
                 + ~~~~~~~~~~~~~~~~~
HistoryId      : 2

Message        : AADSTS50097: Device authentication is required.
                 Trace ID: e0e6689e-7e5d-43b3-be28-a43490025900
                 Correlation ID: 01e262f5-9585-45b8-9820-2a12448900fb
                 Timestamp: 2019-06-14 08:19:08Z
StackTrace     :    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T](Boolean respondToDeviceAuthChallenge)
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.AdalHttpClient.GetResponseAsync[T]()
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendHttpMessageAsync(IRequestParameters requestParameters)
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.SendTokenRequestAsync()
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenByDeviceCodeHandler.SendTokenRequestAsync()
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.CheckAndAcquireTokenUsingBrokerAsync()
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.RunAsync()
                    at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.AcquireTokenByDeviceCodeAsync(DeviceCodeResult deviceCodeResult)
                    at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.DoAcquireToken(AdalConfiguration config, String userId, SecureString password, Action`1 promptAction, Boolean renew)
                    at Microsoft.Azure.Commands.Common.Authentication.UserTokenProvider.SafeAquireToken(AdalConfiguration config, Action`1 promptAction, String userId, SecureString password, Exception& ex)
Exception      : Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException
InvocationInfo : {Connect-AzAccount}
Line           : Connect-AzAccount
Position       : At line:1 char:1
                 + Connect-AzAccount
                 + ~~~~~~~~~~~~~~~~~
HistoryId      : 2

Message        : Response status code does not indicate success: 401 (Unauthorized).
StackTrace     :
Exception      : System.Net.Http.HttpRequestException
InvocationInfo : {Connect-AzAccount}
Line           : Connect-AzAccount
Position       : At line:1 char:1
                 + Connect-AzAccount
                 + ~~~~~~~~~~~~~~~~~
HistoryId      : 2

Message        : {"error":"invalid_grant","error_description":"AADSTS50097: Device authentication is required.\r\nTrace ID: e0e6689e-7e5d-43b3-be28-a43490025900\r\nCorrelation ID: 01e262f5-9585-45b8-9820-2a12448900fb\r\nTimestamp:
                 2019-06-14 08:19:08Z","error_codes":[50097],"timestamp":"2019-06-14 08:19:08Z","trace_id":"e0e6689e-7e5d-43b3-be28-a43490025900","correlation_id":"01e262f5-9585-45b8-9820-2a12448900fb"}: Unknown error
StackTrace     :
Exception      : Microsoft.IdentityModel.Clients.ActiveDirectory.AdalException
InvocationInfo : {Connect-AzAccount}
Line           : Connect-AzAccount
Position       : At line:1 char:1
                 + Connect-AzAccount
                 + ~~~~~~~~~~~~~~~~~
HistoryId      : 2

DEBUG: AzureQoSEvent: CommandName - Resolve-AzError; IsSuccess - True; Duration - 00:00:00.0960681; Exception - ;
DEBUG: Finish sending metric.
DEBUG: 10:20:44 - ResolveError end processing.
DEBUG: 10:20:44 - ResolveError end processing.
```

**Comment by markcowl** [comment]
@J0F3 What browser are you using to bring up the device authentication page, and what are the broswer settings?

Device login *does* work with MFA require (it is required for my login), so I'm curious what the special setting is that is causing a problem in this case.

After you enter the code on the device code login page, you are taken to a page that is controlled by your STS to login, does this page not contain MFA options, or are they just not working?

**Comment by J0F3** [comment]
@markcowl I used different browsers, I tried Chrome, Edge and Internet Explorer. The browser settings are also pretty much default. (no proxy 
 set or so). The part in the browser works actually perfectly, I enter my username get redirected to the ADFS site of my company where I enter the password and finally I make the MFA login which all works without any problems. But then when it says that you can now close your the browser window, PowerShell gives the error noted above. 

The problem seem really not to be the MFA it self, it seems more the Conditional Access. (enforcing MFA is simply the easiest policy you can create to test the behavior). 

I have done also some troubleshooting with Fiddler where I can see that when PowerShell, in the final step, tries to get the access token from the /token endpoint it gets instead the access token an  "invalid_grant" error back. Which leads then to the above described output. 

So maybe it is really more an Azure AD or Conditional Access problem or the Azure PowerShell has to make some other request if Conditional Access is enabled on the tenant , but I am not sure where root of the problem actually is.

**Comment by markcowl** [comment]
@J0F3 The right thing to do in this case is to engage the ActiveDirectory authentication team, as they actually own both ends of this (both the client-side ADAL library and the behavior of the conditional access policy).  

Meanwhile, can you try the preview of the new authentication experience [here](https://www.powershellgallery.com/packages/Az.Accounts/2.0.0-preview)

This uses a vastly updated authentication experience, and may unblock you while we work out the underlying issue.

**Comment by J0F3** [comment]
@markcowl Thank you for the tip with the preview module. That resolve my issue in fact. It works even when enforcing device authentication with -UseDeviceAuthentication.

Do you have a contact at the ActiveDirectory authentication team which I can contact or will you contact them internally?

Thx!

**Comment by markcowl** [comment]
@J0F3 So, this functionality will become GA this summer, so I think we will track this issue along with that GA.  It appears that this is simply not supported for device authentication in ADAL.

**Comment by markcowl** [comment]
@cormacpayne we should be able to resolve this with the new version fo Az.Accounts supporting MSAL

**Comment by J0F3** [comment]
@markcowl Thx, sounds good. I am currently working with the preview version... 👍

**Comment by pedropaulovc** [comment]
Here's another workaround for folks who would rather not install the `Az.Accounts` preview version:

It seems that this issue only affects PowerShell Core. If you run `Connect-AzAccount` in Windows PowerShell "aka legacy PowerShell", it will work fine with Az.Accounts v1.6.6 and the authentication token will also work in PS Core sessions.

**Comment by sanjaiganesh** [comment]
I had the same issue (AADSTS50097: Device authentication is required). Ensuring to take the log-in through MFA or 2FA resolved the issue..

**Comment by J0F3** [comment]
@markcowl Do you have an ETA for the release of the new Az.Accounts module?
When installing the preview you get now the error that the module is not catalog signed:
` The version '2.0.0' of the module 'Az.Accounts' being installed is not catalog signed.`

Thx
Jonas

**Comment by gowdhamankarthikeyan** [comment]
I'm still facing the issue. What is the workaround?

**Comment by J0F3** [comment]
Use the [preview](https://www.powershellgallery.com/packages/Az.Accounts/2.0.0-preview) 2.0 version of the Az.Accounts module.
`Install-Module -Name Az.Accounts -AllowPrerelease -Force -SkipPublisherCheck`

**Comment by markcowl** [comment]
Updated Preview version should be out next month.  Do not have a committed ETA for GA, but expecting by end of spring / early summer.

**Comment by jcravener** [comment]
Even with the preview version of the Az.Accounts module, it's not working.  I get the following error:
Connect-AzAccount: Error occurred when attempting to acquire the common tenant token. Please run 'Connect-AzAccount` again to authenticate.

**Comment by built2order** [comment]
This issue is evident in macOS Powershell 7.0.0.

`Connect-AzAccount: AADSTS50097: Device authentication is required.
Trace ID: 83130c43-ccfe-45c8-9e0a-xxxxxxxxxxxx
Correlation ID: 54d1bb35-7671-4328-b0ba-xxxxxxxxxxxx
Timestamp: 2020-05-05 22:58:32Z: Response status code does not indicate success: 401 (Unauthorized).`

**Comment by isra-fel** [comment]
Az.Accounts 2.0.1-preview was out: https://www.powershellgallery.com/packages/Az.Accounts/2.0.1-preview
We've updated the version of the authentication lib we used. Could you try with it? Thanks

**Comment by jcravener** [comment]
I tried the 2.0.1-preview and it's now working.  Thanks!

**Comment by sushruth** [comment]
When will this fix be out of preview?

**Comment by J0F3** [comment]
Why is this now closed?! It is still not fixed and we are waiting for the release of the new version of the Az.Accounts module since more that a year now! The preview release is only a workaround and seems now even lacking of features. For example the` Invoke-AzRestMethod ` is not in the preview version (only in the 1.9.3 which has still this issue). 

So when can we expect finally the GA of the new version?

**Comment by isra-fel** [comment]
We plan to GA az.accounts late October. It'll be based on Azure.Identity (which is based on MSAL) and all the new features on the stable version will be merged into it.

**Comment by J0F3** [comment]
@isra-fel Ok, thank you. Really looking forward to that happening finally after more than a year...
Two questions though:
Can we track the progress somewhere? 
Why is the Az.Accounts module not on the list of the preview modules? -> #12755

**Comment by isra-fel** [comment]
> Can we track the progress somewhere?

I'm afraid our development progress isn't exposed to public currently. However you can get detailed timeline of upcoming releases in our milestone page: https://github.com/Azure/azure-powershell/milestones (New Az.Accounts will come in next Az major release S177)

> Why is the Az.Accounts module not on the list of the preview modules? -> #12755

There are two types of preview modules -- 

1. Modules just onboarded Azure PowerShell. They haven't finalized their API surface. Their versions are still under 1.0.0 , indicating they haven't GAed. 
1. Modules already GAed such as Az.Accounts, [Az.Storage](https://www.powershellgallery.com/packages/Az.Storage/2.5.2-preview). They release previews just for experimenting features.

The list contains only modules in the first case.

**Comment by J0F3** [comment]
Ah makes sense. Thanks a lot for the explanation!

**Comment by mcurko** [comment]
I had to go back to v. 1.9.5 to make it work!

**Comment by isra-fel** [comment]
Hi @mcurko if you encountered problems in latest Az.Accounts release, could you [file us an issue](https://github.com/Azure/azure-powershell/issues/new?assignees=&labels=needs-triage&template=az-module-bug-report.md&title=) ? Thanks
